### PR TITLE
Report location provider

### PR DESCRIPTION
**Pull request by oliv3**

### CODE REVIEW

File: app/src/main/java/org/traccar/client/DatabaseHelper.java [hunk 1]
```diff
@@ -91,6 +91,11 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         onCreate(db);
     }
 
+    public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS position;");
```
Comment (luke-jr) [inline app/src/main/java/org/traccar/client/DatabaseHelper.java]: Pretty nasty to just delete data... :/

Comment (tananaev) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: How can you downgrade an app?

Comment (oliv3) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: From e.g. https://apkpure.com/traccar-client/org.traccar.client ?

Comment (tananaev) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: I guess you meant to say that you have to manually download APK. I don't think very many people do that.

As for the original comment, ideally database should be empty anyway if you have internet connection. Traccar Client uploads data automatically and removes it from local database.

Comment (oliv3) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: Right, regular Google Play Store users can't downgrad. But as can be seen @luke-jr did, so everything can happen.

Comment (oliv3) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: As for the original comment, I'm also fine with the way upgrading was done.

Comment (luke-jr) [reply app/src/main/java/org/traccar/client/DatabaseHelper.java]: I did it by copying the apk off another phone that hadn't upgraded yet.

**Comment by oliv3** [comment]
A better name ?

**Comment by oliv3** [comment]
Database changes, I'll leave that to you. That's a harsh optimization for sure.

**Comment by oliv3** [comment]
But yes, if the job is only to replay HTTP POSTs, why not just drop Sqlite ?

**Comment by luke-jr** [comment]
"location-source" or something like that maybe?

**Comment by oliv3** [comment]
@luke-jr no, for "setting". I'd be fine with "tuning", because that's what it does. "complicatedHeuristicName" would be too long.

**Comment by oliv3** [comment]
@tananaev ping ?

**Pull request closed without merging**